### DISCUSSION TITLE
Repair beatmap backgrounds feature

### DISCUSSION THREAD

**Discussion by Qwoozle** [ppy/osu#17334] [Ideas]
It would be neat if there was a feature that could redownload beatmap backgrounds. As someone who has used a program to change all of their backgrounds to a dark screen and now regrets it, having a tool to redownload these backgrounds, specifically **deleted/altered** ones would be nice. I am by no means a competent developer but if mass-downloading the backgrounds from the osu! site does not seem feasible, maybe copying the thumbnail image of the map and setting it as the background would work? (if the thumbnail image has not been deleted/altered)

**Comment by frenzibyte** [comment]
Are you requesting this against osu!(lazer)? if so, that's probably out of scope, since lazer does not support mutating backgrounds in the first place, and you'll have to solve this in osu!(stable) instead before importing your beatmaps.

**Comment by peppy** [comment]
The eventual goal in lazer is to allow repairing all files in beatmaps automatically (apart from videos). We're still waiting on server-side components to make this work, and it's not a super high priority but it will happen.
